### ISSUE TITLE
bower cache [non-existent command] throws error instead of responding with unknown command

### ISSUE THREAD

**Issue by ardouglass** [bower/bower#2290]
When typing `bower cache clear` instead of `bower cache clean`, we received an error instead of the `bower EUNKNOWNCMD   Unknown command: whatever` that is displayed in other instances.

```
/usr/local/lib/node_modules/bower/lib/node_modules/q/q.js:155
                throw e;
                      ^
RangeError: Invalid array length
    at Object.assignToHash (/usr/local/lib/node_modules/bower/lib/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:759:22)
    at Object.compile (/usr/local/lib/node_modules/bower/lib/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:113:27)
    at Object.compileChildren (/usr/local/lib/node_modules/bower/lib/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:794:49)
    at Object.compile (/usr/local/lib/node_modules/bower/lib/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:97:10)
    at Object.compileChildren (/usr/local/lib/node_modules/bower/lib/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:794:49)
    at Object.compile (/usr/local/lib/node_modules/bower/lib/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:97:10)
    at compileInput (/usr/local/lib/node_modules/bower/lib/node_modules/handlebars/dist/cjs/handlebars/compiler/compiler.js:516:53)
    at Object.ret [as std/help-cache.std] (/usr/local/lib/node_modules/bower/lib/node_modules/handlebars/dist/cjs/handlebars/compiler/compiler.js:523:18)
    at render (/usr/local/lib/node_modules/bower/lib/util/template.js:20:27)
    at Object.render (/usr/local/lib/node_modules/bower/lib/util/template.js:30:12)
```

**Comment by sheerun** [comment]
Cannot reproduce. Could you provide your node, npm and bower versions?

**Comment by ardouglass** [comment]
Bower: 1.7.9

Tested with multiple node/npm versions:
NPM: 2.15.1 and 3.8.6
Node: 0.12.14 and 5.11.1

**Comment by davidlehn** [comment]
Same as #2264, fixed in #2270.
